Examples/restcomm-olympus-nopod/restcomm-olympus/MainTableViewController.mm

```diff
@@ -286,75 +290,157 @@ - (void)updateConnectivityStatus:(RCDeviceState)state andConnectivityType:(RCDev
+    cell.textLabel.text = [NSString stringWithFormat:@"%@ %@", contact.firstName, contact.lastName];
+    cell.accessoryType = UITableViewCellAccessoryNone;
+    cell.accessoryView = nil;
+    
+    if (contact.phoneNumbers && [contact.phoneNumbers count] > 0){
+        cell.detailTextLabel.text = [contact.phoneNumbers objectAtIndex:0];
+        if ([contact.phoneNumbers count] > 1){
+            UIImage *image = [UIImage imageNamed:@"settings-30x30.png"];
+            UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
+            
+            CGRect frame = CGRectMake(0.0, 0.0, image.size.width, image.size.height);
+            button.frame = frame;
+            [button setBackgroundImage:image forState:UIControlStateNormal];
+            
+            [button addTarget:self action:@selector(checkButtonTapped:event:) forControlEvents:UIControlEventTouchUpInside];
+            cell.backgroundColor = [UIColor clearColor];
+            cell.accessoryView = button;
+        }
+    }
+    
+    return cell;
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
+    [self performSegueWithIdentifier:@"invoke-messages" sender:indexPath];
+}
+
+
+// Override to support editing the table view.
+- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
+    if (editingStyle == UITableViewCellEditingStyleDelete) {
+        // Delete the row from the data source
+        [Utils removeContactAtIndex:(int)indexPath.row];
+        [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
+        
+    } else if (editingStyle == UITableViewCellEditingStyleInsert) {
+        // Adding is handled in the separate screen
+        // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
+    }
+}
+
+- (void)tableView:(UITableView *)tableView accessoryButtonTappedForRowWithIndexPath:(NSIndexPath *)indexPath
 {
     UIStoryboard *storyboard = [UIStoryboard storyboardWithName:[[NSBundle mainBundle].infoDictionary objectForKey:@"UIMainStoryboardFile"] bundle:nil];
-    SettingsTableViewController *settingsViewController = [storyboard instantiateViewControllerWithIdentifier:@"settings-view-controller"];
-    settingsViewController.device = self.device;
+    PhoneNumbersViewController *phoneNumbersViewController = (PhoneNumbersViewController *)[storyboard instantiateViewControllerWithIdentifier:@"phone-numbers"];
+    phoneNumbersViewController.localContact = [Utils contactForIndex: (int)indexPath.row];
+    phoneNumbersViewController.phoneNumbersDelegate = self;
     
-    [self.navigationController pushViewController:settingsViewController animated:YES];
+    phoneNumbersViewController.preferredContentSize = CGSizeMake(150, [phoneNumbersViewController.localContact.phoneNumbers count] * 50);
```

Comment:
Ok, I before popup view controller, I was thinking of the UIAlertController as action sheet, maybe we can go in that direction. Will leave it for now.

Reply:
After discussing with @ipsilantide, here's what I'd like you to do:
1. Lets re-introduce contact editing functionality as we had in the past. The way it will work is that:
   - When user presses the (i) they will be navigated to a new screen very similar to the bottom middle screen, in [this](https://telestax.com/restcomm-client-ios-sdk-beta-3-is-out/) screen shot. Where the CONTACT group should be removed and the ACTIONS should probably be changed to something like NUMBERS, and then instead of a list of audio/video/message items we should have the list of numbers for this contact.
   - Would be nice to see the contact name showing on the top somehow
2. Hook again the functionality for Edit of the contact where for now **we 'll only be able to modify the non-phone contacts** (to avoid creating inconsistencies). Also, let's not allow updating the Name, just the number to avoid messing with hashes identifying entries. Notice that  the old editing code should still be there so maybe it will help you. Same for the ViewController + View of update functionality; it should be in the Storyboard.
3. From the Edit screen the user will also be able to tap on any of the numbers to be navigated to the chat/call screen.

Reply:
Done.